### PR TITLE
jaegerreceiver: Remove use of parsed ports, use address provided by the user

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: receiver/jaegerreceiver/config.go [hunk 1]
```diff
@@ -94,38 +94,36 @@ func (cfg *Config) Validate() error {
 		return fmt.Errorf("must specify at least one protocol when using the Jaeger receiver")
 	}
 
-	var grpcPort int
 	if cfg.GRPC != nil {
-		var err error
-		if grpcPort, err = extractPortFromEndpoint(cfg.GRPC.NetAddr.Endpoint); err != nil {
+		if err := checkPortFromEndpoint(cfg.GRPC.NetAddr.Endpoint); err != nil {
 			return fmt.Errorf("unable to extract port for the gRPC endpoint: %w", err)
```
Comment (codeboten) [inline receiver/jaegerreceiver/config.go:99]: The word "extract" here doesn't make much sense anymore... not sure if it ever would have to an end user :D

File: receiver/jaegerreceiver/factory.go [hunk 2]
```diff
@@ -143,17 +138,17 @@ func createTracesReceiver(
 
 // extract the port number from string in "address:port" format. If the
 // port number cannot be extracted returns an error.
```
Comment (codeboten) [inline receiver/jaegerreceiver/factory.go:140]:
```suggestion
// checkPortFromEndpoint checks that the endpoint string contains a port in the format "address:port". If the
// port number cannot be parsed, returns an error.
```

Comment (codeboten) [inline receiver/jaegerreceiver/factory.go:140]:
```suggestion
// checkPortFromEndpoint checks that the endpoint string contains a port in the format "address:port". If the
// port number cannot be parsed, returns an error.
```

**Review by codeboten — APPROVED** [review]
This is much cleaner, just a couple of comment/suggestion

File: receiver/jaegerreceiver/config.go (hunk 1 above)
Comment (bogdandrutu) [reply receiver/jaegerreceiver/config.go:99]: Done.

File: receiver/jaegerreceiver/factory.go (hunk 2 above)
Comment (bogdandrutu) [reply receiver/jaegerreceiver/factory.go:140]: Done.

Comment (bogdandrutu) [reply receiver/jaegerreceiver/factory.go:140]: Done.

**Pull request merged**